### PR TITLE
fix(agg): Allow non-connected wallets to get swap options

### PR DESCRIPTION
**Pull request by th0rOdinson**

### CODE REVIEW

File: apps/root/src/hooks/useSwapOptions.ts [hunk 1]
```diff
@@ -85,23 +85,23 @@ function useSwapOptions(
...
-              debouncedGasSpeed,
-              debouncedAccount as Address,
-              debouncedChainId,
-              debouncedDisabledDexes,
-              // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-              debouncedIsPermit2Enabled,
-              // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-              debouncedSourceTimeout
-            );
+            const promiseResult = await aggregatorService.getSwapOptions({
```
Comment (0xsambugs) [inline apps/root/src/hooks/useSwapOptions.ts:88]: Really like using the named params 👏

File: apps/root/src/hooks/useTotalGasSaved.ts [hunk 2]
```diff
@@ -37,18 +37,13 @@ function useTotalGasSaved(position: Position | undefined | null): [bigint | unde
...
-            position.rate.amount,
-            undefined,
-            SORT_LEAST_GAS,
-            undefined,
-            undefined,
-            undefined,
-            undefined,
-            position.chainId
-          );
+          const options = await aggregatorService.getSwapOptions({
```
Comment (0xsambugs) [inline apps/root/src/hooks/useTotalGasSaved.ts:40]: Same!

File: apps/root/src/services/aggregatorService.ts [hunk 3]
```diff
@@ -91,28 +91,45 @@ export default class AggregatorService {
...
+    usePermit2?: boolean;
+    sourceTimeout?: TimeoutKey;
+  }) {
     const balance = await this.walletService.getBalance({ account: takerAddress, token: from });
 
-    const isOnNetwork = !chainId || currentNetwork.chainId === chainId;
+    let isOnNetwork = false;
+    try {
+      const currentNetwork = await this.providerService.getNetwork(takerAddress);
+      isOnNetwork = !chainId || currentNetwork.chainId === chainId;
```
Comment (0xsambugs) [inline apps/root/src/services/aggregatorService.ts:128]: It is worse for readability, but if `!chainId`, then you don't need to ask the provider service for the network, and you don't need to wait for it
```suggestion
      isOnNetwork = !chainId || (await this.providerService.getNetwork(takerAddress)).chainId === chainId;
```

**Review by 0xsambugs — APPROVED** [review]
Looks good!

File: apps/root/src/services/aggregatorService.ts (hunk 3 above)
Comment (FiboApe) [reply apps/root/src/services/aggregatorService.ts:128]: Agree

Comment (th0rOdinson) [reply apps/root/src/services/aggregatorService.ts]: Like it!

**Review by FiboApe — APPROVED** [review]

**Pull request merged**